packages/components/src/widgets/LambdaWidget/FunctionChart/utils.ts

```diff
@@ -55,11 +56,19 @@ export function getFunctionImage<T extends SqNumericFnPlot | SqDistFnPlot>(
   environment: Env
 ) {
   const scale = sqScaleToD3(plot.xScale);
+
+  const signatures = plot.fn.signatures();
+  const domain = signatures[0][0]?.domain;
+  const isDateRange = domain?._value.type === "DateRange";
+
   scale.domain([
     plot.xScale?.min ?? functionChartDefaults.min,
     plot.xScale?.max ?? functionChartDefaults.max,
   ]);
 
+  const wrapWithType = (v: number) =>
+    isDateRange ? SqDateValue.fromNumber(v) : SqNumberValue.create(v);
```

Comment:
Hmm. So, if you used real d3 time scales instead of patched linear scales, then this would be easier, because values obtained from the scale would have correct types and you won't have to check the domain type here.

Continuing from my suggestion in another comment about returning `SqValue` values from `SqDistribution` methods, maybe in the future we could implement d3 scales over SqValues. I believe d3 is flexible enough for that. Then we could get rid of all this custom code, and `rangeByCount` would return correctly typed values that we could pass to `plot.fn.call(...)` directly.